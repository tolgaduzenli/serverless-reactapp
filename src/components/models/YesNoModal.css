.modal-view {
    display: flex;
    flex-direction:column;
    justify-content: center;
    align-items: space-between;
}

.button-group{
    display:flex;
    flex-direction:row;
    justify-content: center;
}

.button-group > button{
    margin:15px;
}