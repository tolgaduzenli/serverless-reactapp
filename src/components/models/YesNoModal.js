import React from 'react'
import Modal from 'react-awesome-modal'
import './YesNoModal.css'

class YesNoModal extends React.Component {

    render() {
        const { onClickYes, onClickNo, note, showModal } = this.props
        return <section>
            <Modal
                visible={showModal}
                width="400"
                height="200"
                effect="fadeInUp"
                onClickAway={() => onClickNo()}
            >
                <div className="modal-view">
                    <div>Do you want to remove the note</div>
                    <div>Title: {note.title}</div>
                    <div>Description: {note.description}</div>
                    <div className="button-group">
                        <button className="waves-effect waves-light btn-small green lighten-1"
                            onClick={() => onClickYes(note._id)}>Yes
                        </button>
                        <button className="waves-effect waves-light btn-small grey"
                            onClick={() => onClickNo()}>No
                        </button>
                    </div>
                </div>
            </Modal>
        </section>
    }
}

export default YesNoModal
