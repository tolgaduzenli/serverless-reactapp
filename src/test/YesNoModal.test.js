import React from 'react'
import enzyme from 'enzyme'
import YesNoModal from '../components/models/YesNoModal'

describe('<YesNoModal />', () => {
    let wrapper
    let note = {
        title: "",
        description: ""
    }
    beforeEach(() => {
        wrapper = enzyme.shallow(
            <YesNoModal closeModal={jest.fn()}
                showModal={true}
                onClickModalSave={jest.fn()}
                note={note}
            />
        );
    });
    it('should render correctly', () => expect(wrapper).toMatchSnapshot());

    it('Should render a <section />', () => {
        expect(wrapper.find('section').length).toEqual(1)
    })
    it('Should render a <Modal />', () => {
        expect(wrapper.find('Modal').length).toEqual(1)
    })
})