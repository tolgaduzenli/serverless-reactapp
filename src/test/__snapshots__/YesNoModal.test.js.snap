// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YesNoModal /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <YesNoModal
    closeModal={[MockFunction]}
    note={
      Object {
        "description": "",
        "title": "",
      }
    }
    onClickModalSave={[MockFunction]}
    showModal={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Modal
        effect="fadeInUp"
        height="200"
        onClickAway={[Function]}
        visible={true}
        width="400"
      >
        <div
          className="modal-view"
        >
          <div>
            Do you want to remove the note
          </div>
          <div>
            Title: 
            
          </div>
          <div>
            Description: 
            
          </div>
          <div
            className="button-group"
          >
            <button
              className="waves-effect waves-light btn-small green lighten-1"
              onClick={[Function]}
            >
              Yes
            </button>
            <button
              className="waves-effect waves-light btn-small grey"
              onClick={[Function]}
            >
              No
            </button>
          </div>
        </div>
      </Modal>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="modal-view"
        >
          <div>
            Do you want to remove the note
          </div>
          <div>
            Title: 
            
          </div>
          <div>
            Description: 
            
          </div>
          <div
            className="button-group"
          >
            <button
              className="waves-effect waves-light btn-small green lighten-1"
              onClick={[Function]}
            >
              Yes
            </button>
            <button
              className="waves-effect waves-light btn-small grey"
              onClick={[Function]}
            >
              No
            </button>
          </div>
        </div>,
        "effect": "fadeInUp",
        "height": "200",
        "onClickAway": [Function],
        "visible": true,
        "width": "400",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Do you want to remove the note
            </div>,
            <div>
              Title: 
              
            </div>,
            <div>
              Description: 
              
            </div>,
            <div
              className="button-group"
            >
              <button
                className="waves-effect waves-light btn-small green lighten-1"
                onClick={[Function]}
              >
                Yes
              </button>
              <button
                className="waves-effect waves-light btn-small grey"
                onClick={[Function]}
              >
                No
              </button>
            </div>,
          ],
          "className": "modal-view",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Do you want to remove the note",
            },
            "ref": null,
            "rendered": "Do you want to remove the note",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Title: ",
                "",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Title: ",
              "",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Description: ",
                "",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Description: ",
              "",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="waves-effect waves-light btn-small green lighten-1"
                  onClick={[Function]}
                >
                  Yes
                </button>,
                <button
                  className="waves-effect waves-light btn-small grey"
                  onClick={[Function]}
                >
                  No
                </button>,
              ],
              "className": "button-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Yes",
                  "className": "waves-effect waves-light btn-small green lighten-1",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Yes",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "No",
                  "className": "waves-effect waves-light btn-small grey",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "No",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Modal
          effect="fadeInUp"
          height="200"
          onClickAway={[Function]}
          visible={true}
          width="400"
        >
          <div
            className="modal-view"
          >
            <div>
              Do you want to remove the note
            </div>
            <div>
              Title: 
              
            </div>
            <div>
              Description: 
              
            </div>
            <div
              className="button-group"
            >
              <button
                className="waves-effect waves-light btn-small green lighten-1"
                onClick={[Function]}
              >
                Yes
              </button>
              <button
                className="waves-effect waves-light btn-small grey"
                onClick={[Function]}
              >
                No
              </button>
            </div>
          </div>
        </Modal>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="modal-view"
          >
            <div>
              Do you want to remove the note
            </div>
            <div>
              Title: 
              
            </div>
            <div>
              Description: 
              
            </div>
            <div
              className="button-group"
            >
              <button
                className="waves-effect waves-light btn-small green lighten-1"
                onClick={[Function]}
              >
                Yes
              </button>
              <button
                className="waves-effect waves-light btn-small grey"
                onClick={[Function]}
              >
                No
              </button>
            </div>
          </div>,
          "effect": "fadeInUp",
          "height": "200",
          "onClickAway": [Function],
          "visible": true,
          "width": "400",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                Do you want to remove the note
              </div>,
              <div>
                Title: 
                
              </div>,
              <div>
                Description: 
                
              </div>,
              <div
                className="button-group"
              >
                <button
                  className="waves-effect waves-light btn-small green lighten-1"
                  onClick={[Function]}
                >
                  Yes
                </button>
                <button
                  className="waves-effect waves-light btn-small grey"
                  onClick={[Function]}
                >
                  No
                </button>
              </div>,
            ],
            "className": "modal-view",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Do you want to remove the note",
              },
              "ref": null,
              "rendered": "Do you want to remove the note",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Title: ",
                  "",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Title: ",
                "",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Description: ",
                  "",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Description: ",
                "",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="waves-effect waves-light btn-small green lighten-1"
                    onClick={[Function]}
                  >
                    Yes
                  </button>,
                  <button
                    className="waves-effect waves-light btn-small grey"
                    onClick={[Function]}
                  >
                    No
                  </button>,
                ],
                "className": "button-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Yes",
                    "className": "waves-effect waves-light btn-small green lighten-1",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Yes",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "No",
                    "className": "waves-effect waves-light btn-small grey",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "No",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
